src/main/java/net/minecraftforge/event/entity/EntityEvent.java

```diff
@@ -140,48 +141,93 @@ public boolean didChunkChange()
     public static class Size extends EntityEvent
     {
         private final Pose pose;
-        private final EntityDimensions oldSize;
+        private final EntityDimensions originalSize;
         private EntityDimensions newSize;
-        private final float oldEyeHeight;
-        private float newEyeHeight;
 
-        public Size(Entity entity, Pose pose, EntityDimensions size, float defaultEyeHeight)
+        public Size(Entity entity, Pose pose, EntityDimensions size)
+        {
+            super(entity);
+            this.pose = pose;
+            this.originalSize = size;
+            this.newSize = size;
+        }
+
+        public Pose getPose()
+        {
+            return pose;
+        }
+
+        @Deprecated(forRemoval = true, since = "1.20.1")
+        public EntityDimensions getOldSize()
+        {
+            return getOriginalSize();
+        }
+
+        public EntityDimensions getOriginalSize()
         {
-            this(entity, pose, size, size, defaultEyeHeight, defaultEyeHeight);
+            return originalSize;
         }
 
-        public Size(Entity entity, Pose pose, EntityDimensions oldSize, EntityDimensions newSize, float oldEyeHeight, float newEyeHeight)
+        public EntityDimensions getNewSize()
+        {
+            return newSize;
+        }
+
+        public void setNewSize(EntityDimensions newSize)
+        {
+            this.newSize = newSize;
+        }
+    }
+
+    /**
+     * This event is fired whenever {@link IForgeEntity#getEyeHeightForge(Pose)} gets called.<br>
+     * CAREFUL: This is also fired in the Entity constructor. Therefore the entity(subclass) might not be fully initialized. Check Entity#isAddedToWorld() or !Entity#firstUpdate.<br>
+     * <br>
+     * This event is not {@link Cancelable}.<br>
+     * <br>
+     * This event does not have a result. {@link HasResult}
+     * <br>
+     * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
+     **/
+    public static class EyeHeight extends EntityEvent
```

Comment:
Some methods being moved to this new event from the size event is a breaking change as any references to them on the size event previously no longer exist and will cause method not found errors.

Reply:
Yeah this is a pretty major API breakage. Undoing the break while keeping forge-parity is a PITA though, because we have to undo all the changes to Size and somehow make it work alongside EyeHeight

Reply:
Seems like there definitely are mods affected by the change (see https://github.com/MinecraftForge/MinecraftForge/issues/9659)

Reply:
At the very least we may want to add the methods back deprecate for removal (maybe even add error logging on call) so that mods who didn't test functionality (due to said event having not actually been fired) won't straight up crash, even if they will still be in the situation of pre the events being fixed and firing again of not having expected functionality due to them not testing their mod. (And then with the logging users could actually report said issues to the mods and hopefully get them to fix it and just reference the new event)